client/components/theme/index.jsx

```diff
@@ -62,10 +61,6 @@ const Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
-	shouldComponentUpdate( nextProps ) {
-		return ! isEqual( nextProps.theme, this.props.theme );
```

Comment:
The main rationale for this is that we now _have_ to re-render if the `buttonOptions` prop is changed, which will happen e.g. if a different theme is activated: In some of the options' `hideForTheme` predicates, we're using the `isActiveTheme` selector. So if those predicates change, we definitely want to re-render, or a newly activated theme will have the wrong items in its '...' menu. The same holds in principle true for the `screenshotClickUrl` and `onScreenshotClick`.

I think we mostly need to make sure that `buttonOptions` don't change unnecessarily, and _possibly_ by adding back `shouldComponentUpdate`, with a deep comparison (i.e. another `isEqual`) of `buttonOptions` alongside the one for `theme`, plus some comparisons for the stuff I mentioned above (`screenshotClickUrl`, `onScreenshotClick`, etc).